app/models/targeting.rb

```diff
@@ -38,8 +38,9 @@ def resolve_hosts!
     self.search_query = bookmark.query if dynamic? && bookmark.present?
     self.resolved_at = Time.zone.now
     self.validate!
-    # avoid validation of hosts objects - tey will be loaded for no reason.
-    host_ids = User.as(user.login) { Host.authorized(RESOLVE_PERMISSION, Host).search_for(search_query).pluck(:id) }
+    # avoid validation of hosts objects - they will be loaded for no reason.
+    #   pluck(:id) returns duplicate results for HostCollections
```

Comment:
Since it's part of the explanation on why `pluck(:id)` is there, I'm ok with that :)